src/protocols/secure_channel/CASESession.cpp

```diff
@@ -550,18 +551,16 @@ CHIP_ERROR CASESession::SendSigma2()
     ReturnErrorOnFailure(DRBG_get_bytes(&msg_rand[0], sizeof(msg_rand)));
 
     // Generate an ephemeral keypair
-#ifdef ENABLE_HSM_CASE_EPHEMERAL_KEY
-    mEphemeralKey.SetKeyId(CASE_EPHEMERAL_KEY);
-#endif
-    ReturnErrorOnFailure(mEphemeralKey.Initialize());
+    mEphemeralKey = GetP256KeypairBuilder()->BuildP256KeyPairForEphermalUsage();
```

Comment:
The key is never freed/released. It is leaked

Reply:
Cleared in CASESession::Clear()